source/Menu.cpp

```diff
@@ -644,12 +633,53 @@ namespace Duel6 {
 
         return nullptr;
     }
-
-    void Menu::consumeInputEvents() {
+    int Menu::processEvents(bool single) {
         SDL_Event event;
-        while (SDL_PollEvent(&event)) {
-            // Eat all remaining keyboard events;
+        int result = 0;
+        //TODO This logic duplicates event processing logic in Application. Should be refactored.
```

Comment:
Nejde to teda odsud i z Application hodit do nejake funkce, ktera by se volala z obou mist?